lua/entities/gmod_wire_expression2/core/custom/effects.lua

```diff
@@ -183,6 +183,19 @@ e2function void effect:play(string name)
 	util.Effect(name,this)
 end
 
+e2function number effectCanPlay()
+	return isAllowed(self) and 1 or 0
+end
+
+e2function number effectCanPlay(string name)
+	if not isAllowed(self) then return 0 end
+	if effect_blacklist[name] then return 0 end
+	if hook.Run( "Expression2_CanEffect", name:lower(), self ) == false then return 0 end
```

Comment:
Is this / should this be documented somewhere?

Reply:
I didn't add it, just parity with the old function that already had that function call.

https://github.com/wiremod/wire/commit/da4eff1dc1712d7ff091c1137c9838798e71fb59
Looks like it was already an undocumented list, now just an undocumented hook. Either way this would only really be relevant to developers specifically looking to block E2's effects, so they could just check the source of the function and find this hook.

Could maybe in the future compile these hooks into a wiki page